zproject/computed_settings.py

```diff
@@ -364,6 +364,9 @@ def get_dirs(self) -> List[str]:
     'api_by_ip': [
         (60, 100),
     ],
+    'api_by_remote_server': [
+        (60, 1000),
+    ],
```

Comment:
Are RemoteZulipServers prevented from accessing other endpoints?  i.e. can I register as a remote zulip server (which can be done automatically) and use that to get increased rate limits for spamming?

Reply:
Theoretically, but only on the root subdomain, so it's not clear to me if there are any meaningful endpoints that could be used for spamming with this:
```
def validate_api_key(
    request: HttpRequest,
    role: Optional[str],
    api_key: str,
    allow_webhook_access: bool = False,
    client_name: Optional[str] = None,
) -> Union[UserProfile, "RemoteZulipServer"]:
    # Remove whitespace to protect users from trivial errors.
    api_key = api_key.strip()
    if role is not None:
        role = role.strip()

    # If `role` doesn't look like an email, it might be a uuid.
    if settings.ZILENCER_ENABLED and role is not None and "@" not in role:
        try:
            remote_server = get_remote_server_by_uuid(role)
        except RemoteZulipServer.DoesNotExist:
            raise InvalidZulipServerError(role)
        if api_key != remote_server.api_key:
            raise InvalidZulipServerKeyError(role)

        if get_subdomain(request) != Realm.SUBDOMAIN_FOR_ROOT_DOMAIN:
            raise JsonableError(_("Invalid subdomain for push notifications bouncer"))
```